channels/cliprdr/cliprdr_common.c

```diff
@@ -360,7 +360,7 @@ UINT cliprdr_read_file_contents_request(wStream* s, CLIPRDR_FILE_CONTENTS_REQUES
 	Stream_Read_UINT32(s, request->nPositionHigh); /* nPositionHigh (4 bytes) */
 	Stream_Read_UINT32(s, request->cbRequested);   /* cbRequested (4 bytes) */
 
-	if (Stream_GetRemainingLength(s) >= 4)
+	if (Stream_GetRemainingLength(s) >= sizeof(CLIPRDR_FILE_CONTENTS_REQUEST) - sizeof(CLIPRDR_HEADER) - sizeof(BOOL))
```

Comment:
this is wrong.
the `wStream* s` contains all data read from the wire.
this effectively deactivates `clipDataId` reading in all cases as the `sizeof(CLIPRDR_FILE_CONTENTS_REQUEST) - sizeof(CLIPRDR_HEADER)` have already been read (`Stream_Read_UINT32(s, request->cbRequested);` et al)

Reply:
> this is wrong. the `wStream* s` contains all data read from the wire. this effectively deactivates `clipDataId` reading in all cases as the `sizeof(CLIPRDR_FILE_CONTENTS_REQUEST) - sizeof(CLIPRDR_HEADER)` have already been read (`Stream_Read_UINT32(s, request->cbRequested);` et al)

Sorry, this is my editing error. this should be:  request->common.dataLen  >= sizeof(CLIPRDR_FILE_CONTENTS_REQUEST) - sizeof(CLIPRDR_HEADER) - sizeof(BOOL)

Reply:
FreeRDP Proxy has both client and server roles, so the problem is obvious. However, in any scenario, it is more accurate to use request->common.dataLen to judge.